deals/module/deals.go

```diff
@@ -235,61 +235,125 @@ func (m *Module) Watch(ctx context.Context, proposals []cid.Cid) (<-chan deals.D
 	return ch, nil
 }
 
-// FinalDealRecords returns a list of all finalized storage deals.
-// Records are sorted ascending by activation epoch then timestamp.
-func (m *Module) FinalDealRecords() ([]deals.DealRecord, error) {
-	ret, err := m.store.getFinalDeals()
-	if err != nil {
-		return nil, fmt.Errorf("getting final deals: %v", err)
+// ListStorageDealRecords lists storage deals according to the provided options.
+func (m *Module) ListStorageDealRecords(opts ...deals.ListDealRecordsOption) ([]deals.StorageDealRecord, error) {
+	c := deals.ListDealRecordsConfig{}
+	for _, opt := range opts {
+		opt(&c)
+	}
+	var final []deals.StorageDealRecord
+	if !c.OnlyPending {
+		recs, err := m.store.getFinalDeals()
+		if err != nil {
+			return nil, fmt.Errorf("getting final deals: %v", err)
+		}
+		final = recs
 	}
-	return ret, nil
-}
 
-// PendingDealRecords returns a list of all pending storage deals.
-// Records are sorted ascending by timestamp.
-func (m *Module) PendingDealRecords() ([]deals.DealRecord, error) {
-	ret, err := m.store.getPendingDeals()
-	if err != nil {
-		return nil, fmt.Errorf("getting pending deals: %v", err)
+	var pending []deals.StorageDealRecord
+	if c.IncludePending || c.OnlyPending {
```

Comment:
In these cases, we need to add in pending records.

Reply:
Q: Maybe clearer to change `OnlyPending` to `IncludeFinal`? So both flags have "Include".
Like, this valid combination `c.IncludePending=false` and `c.OnlyPending=true` is worth considering? (that condition wouldn't match above if, and this if)

Reply:
Tough call. My hope was that if you didn't provide any options at all, the default would be some sensible results, in this case, only final deals sorted descending. If we make it `IncludeFinal`, there could still be sensible defaults, but only if the default value for `IncludeFinal` is `true`... That could work but that sort of name + default value combination seems unintuitive to me. Default values of `false` usually are easier to understand.

Reply:
I think the current way it works actually does match the second `if`, but I agree it's confusing that there are two options that seemingly contradict each other. The current rule is that the `OnlyPending` wins out, but that would have to be explained.

Reply:
So I'm thinking the less confusing of the options is what you're suggesting: `IncludeFinal` with a default of `true`. I'll give it a try